packages/labs/src/select.tsx

```diff
@@ -0,0 +1,231 @@
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import {
+    AbstractComponent,
+    Button,
+    Classes,
+    HTMLInputProps,
+    IInputGroupProps,
+    InputGroup,
+    IPopoverProps,
+    Keys,
+    Popover,
+    Position,
+    Utils,
+} from "@blueprintjs/core";
+import { IInputListRendererProps, IListItemsProps, InputList } from "./inputList";
+
+export interface ISelectProps<T> extends IListItemsProps<T> {
+    /**
+     * Whether the dropdown list can be filtered.
+     * Disabling this option will remove the `InputGroup` and ignore `inputProps`.
+     * @default true
+     */
+    filterable?: boolean;
+
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (item: T, isActive: boolean, onClick: React.MouseEventHandler<HTMLElement>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
+
+    /**
+     * Props to spread to `InputGroup`. All props are supported except `ref` (use `inputRef` instead).
+     * If you want to control the filter input, you can pass `value` and `onChange` here
+     * to override `Select`'s own behavior.
+     */
+    inputProps?: IInputGroupProps & HTMLInputProps;
+
+    /** Props to spread to `Popover`. Note that `content` cannot be changed. */
+    popoverProps?: Partial<IPopoverProps> & object;
+
+    /**
+     * Whether the filtering state (query string and active item) should be
+     * reset to initial values when an item is selected.
+     * @default false
+     */
+    resetOnSelect?: boolean;
+}
+
+export interface ISelectState<T> {
+    activeItem?: T;
+    isOpen?: boolean;
+    query?: string;
+}
+
+@PureRender
+export class Select<T> extends AbstractComponent<ISelectProps<T>, ISelectState<T>> {
+    public static displayName = "Blueprint.Select";
+
+    public static ofType<T>() {
+        return Select as new () => Select<T>;
+    }
+
+    public state: ISelectState<T> = { isOpen: false, query: "" };
+
+    private DInputList = InputList.ofType<T>();
+    private inputList: InputList<T>;
+    private refHandlers = {
+        inputList: (ref: InputList<T>) => {
+            this.inputList = ref;
+            (window as any).inputList = ref;
+        },
+    };
+    private previousFocusedElement: HTMLElement;
+
+    public render() {
+        // omit props specific to this component, spread the rest.
+        // TODO: should InputList just support arbitrary props? could be useful for re-rendering
```

Comment:
what do you think about the question? should it just support `[key: string]: any;` so you can spread arbitrary things?